From c34533eb023a7586d8d05e577b0808dc22a0d2f6 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 18 Dec 2017 18:57:48 +0100
Subject: [PATCH 1/1] efi_loader: provide link between devices and EFI handles

U-Boot devices and EFI handles can be related, e.g. an
IDE disk relates to a handle with the EFI_BLOCK_IO_PROTOCOL.
Provide a pointer to store this link.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi_loader.h          | 2 ++
 lib/efi_loader/efi_boottime.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/include/efi_loader.h b/include/efi_loader.h
index 2db4e9c09b..bb1e177ae3 100644
--- a/include/efi_loader.h
+++ b/include/efi_loader.h
@@ -139,6 +139,8 @@ struct efi_object {
 	struct list_head protocols;
 	/* The object spawner can either use this for data or as identifier */
 	void *handle;
+	/* Device */
+	struct udevice *dev;
 };
 
 /**
diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 5a3349ecb2..abb798ff34 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -362,6 +362,7 @@ efi_status_t efi_create_handle(efi_handle_t *handle)
 			      (void **)&obj);
 	if (r != EFI_SUCCESS)
 		return r;
+	memset(obj, 0, sizeof(struct efi_object));
 	efi_add_handle(obj);
 	*handle = obj->handle;
 	return r;
-- 
2.14.2

